src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestChannelInternal.java

```diff
@@ -128,41 +111,19 @@
       OpenChannelRequest.OnErrorOption onErrorOption,
       Constants.BdecVersion bdecVersion,
       BufferAllocator allocator) {
-    this.channelName = name;
-    this.dbName = dbName;
-    this.schemaName = schemaName;
-    this.tableName = tableName;
-    this.offsetToken = offsetToken;
-    this.channelSequencer = channelSequencer;
-    this.rowSequencer = new AtomicLong(rowSequencer);
-    this.isValid = true;
     this.isClosed = false;
     this.owningClient = client;
     this.allocator = allocator;
-    this.rowBuffer = createRowBuffer(bdecVersion);
-    this.encryptionKey = encryptionKey;
-    this.encryptionKeyId = encryptionKeyId;
-    this.onErrorOption = onErrorOption;
-    logger.logInfo("Channel={} created for table={}", this.channelName, this.tableName);
-  }
-
-  private RowBuffer<T> createRowBuffer(Constants.BdecVersion bdecVersion) {
-    // TODO: The circular dependency SnowflakeStreamingIngestChannelInternal <-> RowBuffer
-    // (SNOW-657667)
-    // can be probably reconsidered
-    switch (bdecVersion) {
-      case ONE:
-        //noinspection unchecked
-        return (RowBuffer<T>)
-            new ArrowRowBuffer((SnowflakeStreamingIngestChannelInternal<VectorSchemaRoot>) this);
-      case THREE:
-        //noinspection unchecked
-        return (RowBuffer<T>)
-            new ParquetRowBuffer((SnowflakeStreamingIngestChannelInternal<ParquetChunkData>) this);
-      default:
-        throw new SFException(
-            ErrorCode.INTERNAL_ERROR, "Unsupported BDEC format version: " + bdecVersion);
-    }
+    this.channelContext =
+        new ChannelContext(
+            name, dbName, schemaName, tableName, channelSequencer, encryptionKey, encryptionKeyId);
+    AbstractRowBuffer.InitialisationContext bufferConfig =
+        new AbstractRowBuffer.InitialisationContext(
+            allocator, bdecVersion, getFullyQualifiedName(), this::collectRowSize);
+    this.channelState = new ChannelState(onErrorOption, offsetToken, rowSequencer, true);
```

Comment:
Could we try to minimum the number of static classes used? Maybe just create one ChannelProperties class and put everything there? I find it confusing by these fields as a person who reads the code and it's not clear the exact purpose of each one, and if I need to add a new field i'm not sure which one to use

Reply:
The ChannelState is something that is used and mutated by both channel and buffer. I do not want to mix anything else into this because otherwise it is hard to just about what state channel and buffer are coupled with. Btw, just realised that `onErrorOption` actually does not belong to it, I was just trying to somehow keep with the channel but it is not used by channel class.

InitialisationContext is something that is needed only for the buffer.
It is mostly to avoid constructors and methods with a lot of arguments. I can remove it and pass again everything to the constructor.

Reply:
The ChannelState is something that is used and mutated by both channel and buffer. I do not want to mix anything else into this because otherwise it is hard to just about what state channel and buffer are coupled with. Btw, just realised that `onErrorOption` actually does not belong to it, I was just trying to somehow keep with the channel but it is not used by channel class.

InitialisationContext is something that is needed only for the buffer.
It is mostly to avoid constructors and methods with a lot of arguments. I can remove it and pass again everything to the constructor.